k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -7,12 +7,42 @@ use crate::common::{
 };
 use k8s_openapi::api::core::v1::Pod;
 use kube::{api::ObjectList, ResourceExt};
+use openapi::models::{Volume, VolumeStatus};
 use semver::{Version, VersionReq};
 use snafu::ResultExt;
+use std::time::Duration;
 use tracing::{info, warn};
 
 /// Function to check for any volume rebuild in progress across the cluster
 pub(crate) async fn is_rebuilding(rest_client: &RestClientSet) -> Result<bool> {
+    loop {
+        let unhealthy_volumes = list_unhealthy_volumes(rest_client).await?;
+        if unhealthy_volumes.is_empty() {
+            break;
+        }
+
+        for volume in unhealthy_volumes.iter() {
+            match replica_rebuild_count(volume.clone()).await {
+                0 => {
+                    for _i in 0 .. 11 {
+                        // wait for a minute for any rebuild to start
+                        tokio::time::sleep(Duration::from_secs(60_u64)).await;
```

Comment:
etcd cache period is by default 30s. So objects are update with that interval. So i think we should use lesser sleep . perhaps for eg 35s?

Reply:
We return here does the caller of this fn call this again until it gets Ok(false) ?

Reply:
Addressed , changed it to 35seconds , but will need some more upvote on this. @tiagolobocastro

Reply:
Problem is now we wait 5.8 minutes rather than 10 minutes..
Either change back to 60 or adjust loop to something like:
```rust
let start = std::time::Instant::now();
const rebuildMaxWait: Duration = Duration::from_seconds(60); // -> can put this together to other consts/statics.
while start.elapased() < rebuildMaxWait {
...
}
```

Reply:
reverted to 60 seconds interval.